source/dub/compilers/ldc.d

```diff
@@ -241,7 +237,18 @@ config    /etc/ldc2.conf (x86_64-pc-linux-gnu)
 			case TargetType.executable: break;
 			case TargetType.library:
 			case TargetType.staticLibrary:
-				settings.addDFlags("-lib");
+				// -oq: name object files uniquely (so the files don't collide)
+				settings.addDFlags("-lib", "-oq");
+				// -cleanup-obj (supported since LDC v1.1): remove object files after archiving to static lib
+				if (platform.frontendVersion >= 2071) {
+					settings.addDFlags("-cleanup-obj");
+				}
+				if (platform.frontendVersion >= 2095) {
+					// since LDC v1.25, -cleanup-obj additionally defaults to a unique temp -od directory
+				} else {
+					// resort to a unique-ish -od directory
+					settings.addDFlags("-od=" ~ settings.targetPath ~ "/obj");
+				}
```

Comment:
```suggestion
				if (platform.frontendVersion < 2095) {
					// Since LDC v1.25, -cleanup-obj defaults to a unique temp -od directory
					// We need to resort to a unique-ish -od directory before that
					settings.addDFlags("-od=" ~ settings.targetPath ~ "/obj");
				}
```